lints/lint_sub_cert_valid_time_too_long.go

```diff
@@ -18,16 +19,32 @@ func (l *subCertValidTimeTooLong) CheckApplies(c *x509.Certificate) bool {
 }
 
 func (l *subCertValidTimeTooLong) Execute(c *x509.Certificate) *LintResult {
-	if c.NotBefore.AddDate(0, 39, 0).Before(c.NotAfter) {
-		return &LintResult{Status: Error}
+	july_1_2016 := time.Date(2016, time.July, 1, 0, 0, 0, 0, time.UTC)
+	march_1_2018 := time.Date(2018, time.March, 1, 0, 0, 0, 0, time.UTC)
+	if c.NotBefore.Before(july_1_2016) {
+		// Vacuously passed
+		return &LintResult{Status: Pass}
+	}
+	if c.NotBefore.After(march_1_2018) {
+		if c.NotBefore.AddDate(0, 0, 825).Before(c.NotAfter) {
+			// NotAfter - NotBefore > 825 days
+			return &LintResult{Status: Error}
+		} else {
+			return &LintResult{Status: Pass}
+		}
+	} else {
+		// Between 2016/July/01 and 2018/March/01: 39 months
+		if c.NotBefore.AddDate(0, 39, 0).Before(c.NotAfter) {
+			return &LintResult{Status: Error}
+		}
+		return &LintResult{Status: Pass}
 	}
-	return &LintResult{Status: Pass}
 }
 
 func init() {
 	RegisterLint(&Lint{
 		Name:          "e_sub_cert_valid_time_too_long",
-		Description:   "CAs MUST NOT issue subscriber certificates with validity periods longer than 39 months regardless of circumstance.",
+		Description:   "Subscriber Certificates issued after 1 March 2018 MUST have a Validity Period no greater than 825 days. Subscriber Certificates issued after 1 July 2016 but prior to 1 March 2018 MUST have a Validity Period no greater than 39 months. ",
 		Citation:      "BRs: 6.3.2",
 		Source:        CABFBaselineRequirements,
 		EffectiveDate: util.SubCert39Month,
```

Comment:
This will make this only run for certs issued after this time, which probably also needs to be changed